3i-deploy/recipes/deploy-nodejs-CLI.rb

```diff
@@ -58,6 +58,12 @@
     action :delete
   end
 
+  execute 'build' do
+    command "cd #{deploy[:deploy_to]}/current && npm run build"
+    only_if { deploy['environment_variables']['REQUIRES_BUILD'] == 'true' }
+    user 'root'
```

Comment:
I'm trying to think if there would be any undesirable side-effects from running as root.  `npm` couldn't even write files to the directory from which it was run?  Seems weird, but should be fine. Maybe something to do with how opsworks is setup? What user were we before we switch it to `root` here?

Reply:
> I'm trying to think if there would be any undesirable side-effects from running as root.

That's a very valid concern (and was mine too), but if I remember correctly I was painted into a corner on this one. I don't believe the build command has sufficient permissions to write to the source directory without root, since some of what set up that directory in other recipes was root.

> What user were we before we switch it to root here?

This is where I'm a bit fuzzy on chef too, but I don't believe we _are_ necessarily a user before this command. In most of the other recipes I saw when a shell command was executed the user was specified each time. Most of the time the user specified is either root or ubuntu.